lib/LedgerSMB/Template/XLS.pm

```diff
@@ -0,0 +1,255 @@
+            $properties{$attr} = $val;
+        } elsif ($attr eq 'num_format') {
+            $properties{'num_format'} = $val;
+        } else {
+            warn $attr;
+        }
+        $format->del_att($attr);
+    }
+    $_->set_att(type => 'format', format => { %properties });
+}
+
+=pod
+# Not yet implemented
+sub _named_format {
+    my ($name, $t, $format) = @_;
+#warn "_named_format:" . p($format);
+    $format->{att}{$name} = 1;
+    &_format_handler($t, $format);
+    $format->set_att(type => 'named_format');
+}
+=cut
+
+sub _format_cleanup_handler {
+    my ($t, $format) = @_;
+}
+
+sub _xls_process {
+    my ($filename, $template) = @_;
+
+    $workbook  = Spreadsheet::WriteExcel->new("$filename");
+
+    my $parser = XML::Twig->new(
+        start_tag_handlers => {
+            worksheet => \&_worksheet_handler,
+            row => \&_row_handler,
+            cell => \&_cell_handler,
+            formula => \&_formula_handler,
+            format => \&_format_handler,
+=pod
+            bold => sub { &_named_format('bold', @_) },
+            hidden => sub { &_named_format('hidden', @_) },
+            italic => sub { &_named_format('italic', @_) },
+            shadow => sub { &_named_format('shadow', @_) },
+            strikeout => sub { &_named_format('strikeout', @_) },
+=cut
+            },
+        twig_handlers => {
+            format => \&_format_cleanup_handler,
+            bold => \&_format_cleanup_handler,
+            hidden => \&_format_cleanup_handler,
+            italic => \&_format_cleanup_handler,
+            shadow => \&_format_cleanup_handler,
+            strikeout => \&_format_cleanup_handler,
+            }
+        );
+    $parser->parse($template);
+    handle_subtree($parser->root);
+    #$parser->purge;
+    $workbook->close;
+}
+
+use Switch;
+
+sub handle_subtree {
+    my ($tree,$format) = @_;
+    my @children = $tree->children;
+    foreach my $child (@children) {
+        my $att = $child->{att};
+        switch ($att->{type}) {
+            case 'worksheet' {
+                $worksheet = $workbook->add_worksheet($att->{name});
+                handle_subtree($child);
+            }
+            case 'cell' {
+                $worksheet->write($att->{row},$att->{col},$att->{text},$format);
+            }
+            case 'format' {
+                my $format = $workbook->add_format(%{$att->{format}});
+                handle_subtree($child,$format);
+            }
+            case 'row' {
+                handle_subtree($child,$format);
+            }
+            else { warn p($child); }
+        }
+        $child->purge;
+    }
+}
+
+sub get_template {
+    my $name = shift;
+    return "${name}.${extension}t";
+}
+
+sub preprocess {
+    my $rawvars = shift;
+    return LedgerSMB::Template::_preprocess($rawvars, \&escape);
+}
+
+sub escape {
+    my $vars = shift @_;
+    return undef unless defined $vars;
+    $vars = escapeHTML($vars);
```

Comment:
Maybe add a comment inthe code as to why we need to html escape in an xls file...